### ISSUE TITLE
Returning an error in a test does not fail the test

### ISSUE THREAD

**Issue by mcastorina** [vlang/v#10325]
**V version:** V 0.2.2 a368800
**OS:** linux

**What did you do?**
```v
fn example() ? { return error('oh no') }

fn test_example() ? {
	example() or { return error('oh no') }
}
```

```
v test main_test.v
```

**What did you expect to see?**
Failed test
 
**What did you see instead?**
All tests pass

**Comment by spytheman** [comment]
Good find.

**Comment by spytheman** [comment]
Should be fixed in 5400a76:
![image](https://user-images.githubusercontent.com/26967/120516170-bd861c80-c3d7-11eb-88db-ee52200d427d.png)
